src/app/util/ember-compatibility-functions.cpp

```diff
@@ -575,36 +556,66 @@ CHIP_ERROR WriteSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVReader & a
     attributePathParams.mClusterId   = aClusterInfo.mClusterId;
     attributePathParams.mAttributeId = aClusterInfo.mAttributeId;
 
+    Protocols::InteractionModel::Status status = Protocols::InteractionModel::Status::Failure;
+
     // Named aPath for now to reduce the amount of code change that needs to
     // happen when the above TODO is resolved.
     ConcreteDataAttributePath aPath(aClusterInfo.mEndpointId, aClusterInfo.mClusterId, aClusterInfo.mAttributeId);
-    const EmberAfAttributeMetadata * attributeMetadata =
-        emberAfLocateAttributeMetadata(aPath.mEndpointId, aPath.mClusterId, aPath.mAttributeId, CLUSTER_MASK_SERVER, 0);
-
-    if (attributeMetadata == nullptr)
     {
-        return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::UnsupportedAttribute);
-    }
+        AttributeAccessInterface * attrOverride = findAttributeAccessOverride(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+        if (attrOverride != nullptr)
```

Comment:
This looks like a bad merge.  It lost the "check that this attribute even exists before trying to write to it" fix.  I would have expected out tests to catch that; did they not?

Reply:
I redid the refactoring over latest master branch, and tests pass now. They weren't passing before, but I didn't dig into why, since it was easier to reapply the refactoring on latest source code.